tests/SpecTests/Operation.php

```diff
@@ -500,6 +504,35 @@ private function executeForSession(Session $session, Context $context)
         }
     }
 
+    private function executeForTestRunner(FunctionalTestCase $test, Context $context)
+    {
+        switch ($this->name) {
+            case 'assertSessionPinned':
+                $session = $context->{$this->arguments['session']};
+                $test->assertNotNull($session->getServer());
```

Comment:
I realize it's just as effective in practice, but would an `assertInstanceOf` check be more appropriate here?

Reply:
It's just as effective if we rely on the return type of `getServer`. Since this isn't strict at all, it's better to be defensive here and use `assertInstanceOf`. Good catch! 👍